### PR TITLE
PT-1141: vc-build improvements

### PR DESCRIPTION
**Pull request by krankenbro**
### Added
- Targets: Init, Install, InstallPlatform, InstallModules,, Uninstall

### CODE REVIEW

**Review by akak1977 — APPROVED** [review]

File: build/PlatformTools/GithubManager.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            if (string.IsNullOrEmpty(releaseTag)) {
+               release = await client.Repository.Release.GetLatest(GithubUser, PlatformRepo);
+            } else
+            {
+               release = await client.Repository.Release.Get(GithubUser, PlatformRepo, releaseTag);
+            }
+            return release;
+        }
+
+        public static Tuple<string, string> GetRepoFromUrl(string url)
```
Comment (Vectorfield4) [inline build/PlatformTools/GithubManager.cs:28]: Add XML doc what does first and the second parameter means.
``` csharp
/// <summary>
/// Do something
/// </summary>
```

File: build/PlatformTools/GithubManager.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            Release release;
+            if (string.IsNullOrEmpty(releaseTag))
+            {
+                release = await client.Repository.Release.GetLatest(GithubUser, moduleRepo);
+            }
+            else
+            {
+                release = await client.Repository.Release.Get(GithubUser, moduleRepo, releaseTag);
+            }
+            return release;
```
Comment (Vectorfield4) [inline build/PlatformTools/GithubManager.cs:47]:
```suggestion
            return string.IsNullOrEmpty(releaseTag)
                ? await client.Repository.Release.GetLatest(GithubUser, moduleRepo)
                : await client.Repository.Release.Get(GithubUser, moduleRepo, releaseTag);
```

File: build/PlatformTools/LocalModuleCatalog.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                    RefreshProbingFolderOnStart = true,
+                    DiscoveryPath = discoveryPath,
+                    ProbingPath = probingPath,
+                };
+                var options = Microsoft.Extensions.Options.Options.Create<LocalStorageModuleCatalogOptions>(moduleCatalogOptions);
+                var logger = new LoggerFactory().CreateLogger<LocalStorageModuleCatalog>();
+                _catalog = new LocalStorageModuleCatalog(options, logger);
+                _catalog.Load();
+            }
+            _catalog.Reload();
```
Comment (Vectorfield4) [inline build/PlatformTools/LocalModuleCatalog.cs:29]: Any reason to Load/Reload operation sequentially?

**Review by Vectorfield4 — CHANGES_REQUESTED** [review]

File: build/PlatformTools/Build.PackageManager.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    private string GetDiscoveryPath()
+    {
+        var configuration = AppSettings.GetConfiguration(RootDirectory, AppsettingsPath);
+        return string.IsNullOrEmpty(DiscoveryPath) ? configuration.GetModulesDiscoveryPath() : DiscoveryPath;
+    }
+
+    private bool NeedToInstallPlatform(string version)
+    {
+        bool result = true;
+        var platformWeb = RootDirectory / "VirtoCommerce.Platform.Web.dll";
```
[the remark this answers is not in the document]
Comment (akak1977) [reply build/PlatformTools/Build.PackageManager.cs:89]: No need I think. It's normal way for NUKE.

File: build/PlatformTools/Build.PackageManager.cs [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+            }
+        }
+        PackageManager.ToFile(packageManifest);
+    });
+
+    Target InstallPlatform => _ => _
+    .Executes(async () =>
+    {
+        var packageManifest = PackageManager.FromFile(PackageManifestPath);
+        var platformZip = TemporaryDirectory / "platform.zip";
```
[the remark this answers is not in the document]
Comment (akak1977) [reply build/PlatformTools/Build.PackageManager.cs:72]: No need I think. It's normal way for NUKE.

[the remark this answers is not in the document]
Comment (Vectorfield4) [reply build/PlatformTools/Build.PackageManager.cs:72]: Yes, I forgot about `/` operator. Thanks.

File: build/PlatformTools/Build.PackageManager.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vectorfield4) [reply build/PlatformTools/Build.PackageManager.cs:89]: Yes, I forgot about `/` operator. Thanks.

**Review by Vectorfield4 — APPROVED** [review]

**Review by tatarincev — APPROVED** [review]

**Pull request merged**